sdk/servicebus/service-bus/src/connectionContext.ts

```diff
@@ -329,29 +329,14 @@ export namespace ConnectionContext {
         await delay(Constants.connectionReconnectDelay);
 
         const detachCalls: Promise<void>[] = [];
-
-        // Call onDetached() on sender so that it can gracefully shutdown
-        for (const senderName of Object.keys(connectionContext.senders)) {
-          const sender = connectionContext.senders[senderName];
-          if (sender) {
-            logger.verbose(
-              "[%s] calling detached on sender '%s'.",
-              connectionContext.connection.id,
-              sender.name
-            );
-            detachCalls.push(
-              sender.onDetached().catch((err) => {
-                logError(
-                  err,
-                  "[%s] An error occurred while calling onDetached() the sender '%s': %O.",
-                  connectionContext.connection.id,
-                  sender.name,
-                  err
-                );
-              })
-            );
-          }
-        }
+        // Neither we do recovery for the sender, nor we cleanup
+        // No recovery:
+        //   Because we don't want to keep the sender active all the time
+        //   and the "next" send call would bear the burden of creating the link
+        // No cleanup:
+        //   "Closing the link" cleanup would step over new link initializations
+        //   and can possibly clear the link once created, hence we do the cleanup
+        //   at the time of new link creation
```

Comment:
Take the case of a disconnected event firing and user making no more send requests. In that case, the loop to keep renewing the token on the sender link does not get cleared and we will have a rogue loop on our hands

Reply:
I realize that we can argue that in the normal case of user stopping to make any send requests and no disconnected event, our loop will still keep trying to renew the lock. In that case, at some point the service will close the link because it has been idle for too long. That results in an "amqp close" event in which case we should clean up the timers and assets from our end. In this PR, we are removing those as well

Reply:
Upon discussion, reverted this in favour of the proposal for cleanup with detached where onDetached would be called before the refreshConnection() is invoked.

Triggered live tests https://dev.azure.com/azure-sdk/internal/_build/results?buildId=567200&view=results